ember-toucan-core/src/components/form/file-input-field.ts

```diff
@@ -0,0 +1,94 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+
+type FileTarget = EventTarget & { files?: FileList };
+export type FileEvent = (Event | MouseEvent) & { target: FileTarget | null };
+
+interface ToucanFormFileInputFieldComponentSignature {
+  Element: HTMLInputElement;
+  Args: {
+    accept?: string;
+    deleteLabel: string;
+    error?: string;
+    files?: File[];
+    hint?: string;
+    label: string;
+    trigger: string;
+    isDisabled?: boolean;
+    multiple?: boolean;
+    onChange?: (files: File[], event: FileEvent) => void;
+    rootTestSelector?: string;
+  };
+}
+
+export default class ToucanFormFileInputFieldComponent extends Component<ToucanFormFileInputFieldComponentSignature> {
+  constructor(
+    owner: unknown,
+    args: ToucanFormFileInputFieldComponentSignature['Args']
+  ) {
+    assert(
+      'A "@label" argument is required for Form::FileInput::Field',
+      args.label !== undefined
+    );
+
+    assert(
+      'A "@deleteLabel" argument is required for Form::FileInput::Field. This provides an accessible label for the delete button.',
+      args.deleteLabel !== undefined
+    );
+
+    assert(
+      'A "@trigger" argument is required for FileInputField, this prompts the user to select files',
+      args.trigger !== undefined
+    );
+
+    super(owner, args);
+  }
+
+  get accept() {
+    return this.args.accept ?? '*';
+  }
+
+  get multiple() {
+    return this.args.multiple ?? true;
+  }
+
+  formatSize(size: number) {
+    return `${Math.round(size / 1000)} KB`;
+  }
+
+  get hasError() {
+    return Boolean(this.args?.error);
+  }
+
+  @action
+  onChange(event: FileEvent) {
+    if (!event.target?.files) {
+      return;
+    }
+
+    const files = [...event.target.files];
```

Comment:
Something is wrong here (beyond `@onChange={{@onChange}}`, see my other comment). In the docs app I tried to select multiple files, but once I select a new file, it _replaces_ the existing one.

I think this should be something like this probably, to merge the new file(s) with the existing ones?

```suggestion
    const files = [...this.args.files, ...event.target.files];
```

But only when `this.args.multiple === true`, otherwise we indeed need to replace.

And we should add a test for multiple selection, to cover this!

Reply:
Thank you for catching this!

1. new test for `multiple` has been added. Covers `adding` (`multiple`) and `replacing` (no `multiple` arg).
2. this whole `this.onChange` has been deleted from `file-input-field` (see explanation above...it wasn't in use)
3. the logic you suggested is now in `controls/file-input`.